### ISSUE TITLE
How to read bcf file ?

### ISSUE THREAD

**Issue by bioinfornatics** [samtools/htsjdk#1537]
Dear,

Firstly thanks for your amazing work on this library.

I try to read a bcffile through `VCFFileReader` but I have this error:

```
Unable to parse header with error: Input stream does not contain a BCF encoded file; BCF magic header info not found, at record 0 with position 0:, for input source: file:///somewhere/build/resources/test/dataset1.bcf
```

This file is valid and I am able to read it with `pysam` in python

```
$ bcftools  view -h /somewhere/build/resources/test/dataset1.bcf | head | grep fileformat
##fileformat=VCFv4.1
```

I use **htsjdk:** `2.23.0` **java:** `8+`

Thanks for your help

**Comment by cmnbroad** [comment]
@bioinfornatics Oh - can you run `od -c -N 30 <file>` and include the results here. You may be hitting this embarrassing bug which we've never fixed https://github.com/samtools/htsjdk/issues/946.

**Comment by bioinfornatics** [comment]
@cmnbroad thanks I was trying to get the magic number as it is done into [BCFVersion.java](https://github.com/samtools/htsjdk/blob/e610af80b13a4cd9540f5aeb37fd44577b0f5754/src/main/java/htsjdk/variant/bcf2/BCFVersion.java) with `hexdump -C` but I had any idea to do next :-)

So the result of `od -c -N 30`
```
0000000 037 213  \b 004  \0  \0  \0  \0  \0 377 006  \0   B   C 002  \0
0000020 250 003 255 224 313 213 023   I 034 307 333   L 226 245
```

**Comment by cmnbroad** [comment]
Yes, that looks like a gzip header. Sadly, htsjdk lacks support for gzipped BCF, as mentioned above.

**Comment by lindenb** [comment]
@bioinfornatics you can try to use a VCFIteratorBuilder instead of a VCFFileReader . As far as I remember, the GZ compression was first tested. (But you cannot do random-access)

**Comment by bioinfornatics** [comment]
thanks @lindenb and @cmnbroad

**Comment by tfenne** [comment]
Closing as stale — sorry this went unanswered for so long.

Two known limitations are the likely cause of that "BCF magic header info not found" error: htsjdk can't read block-gzipped BCF (#946), and it rejects BCF headers containing `IDX` fields (#507), both of which bcftools and pysam handle. Those issues track the real work.

If you're still hitting this on a current release, please open a fresh issue with the file.
